horreum-backend/src/test/java/io/hyperfoil/tools/horreum/svc/RunServiceTest.java

```diff
@@ -147,11 +138,13 @@ public void testTransformationWithNestedSchema(TestInfo info) throws Interrupted
       Schema roadRunnerSchema = createExampleSchema("RoadRunnerCorp", "RoadRunnerInc", "RoadRunnerRrUs", false);
       assertNotNull(roadRunnerSchema.id);
 
-      String foo = "foo";
-      NamedJsonPath acmePath = createExampleNamedJsonPath(foo, "$.value");
-      String function = "({ "+foo+" }) => { return ({  \"outcome\": { "+foo+" } }) }";
+      String variable = "foo";
+      NamedJsonPath acmePath = createExampleNamedJsonPath(variable, "$.value");
+      String function = "({ "+variable+" }) => { return JSON.stringify(({  \"outcome\": { "+variable+" } })) }";
```

Comment:
I needed to use the function `JSON.stringify` to get the output correctly quoted. Without that call the output is not valid JSON for the Jackson library to accept. It requires the user to use this function. I do wonder if the code should make the users life easier. By adding a second JS processing layer to run `JSON.stringify` on the output. WDYT ?

Reply:
I guess that this might be caused by the `.toString()` in https://github.com/Hyperfoil/Horreum/pull/160/files#diff-61d65ffe5339fc552d73ec29865f9c8298ea43c9f154eb226f3d953bd52e867dR1127 and the attempt to parse the object back from string - `JSON.stringify` shouldn't be used if the function is supposed to return an object.

Reply:
I didn't see that being the cause. The outcome seen in this [comment](https://github.com/Hyperfoil/Horreum/pull/160#discussion_r839352281) is what happens without `JSON.stringify`.

Reply:
When I fetch your branch, and remove the `JSON.stringify()` I get this:
```
2022-04-05 16:59:06,238 ERROR [io.hyp.too.hor.svc.Util] (executor-thread-2) Failed to parse into JSON: {outcome: {foo: 42}}: com.fasterxml.jackson.core.JsonParseException: Unexpected character ('o' (code 111)): was expecting double-quote to start field name
 at [Source: (String)"{outcome: {foo: 42}}"; line: 1, column: 3]
	at com.fasterxml.jackson.core.JsonParser._constructError(JsonParser.java:2337)
	at com.fasterxml.jackson.core.base.ParserMinimalBase._reportError(ParserMinimalBase.java:710)
	at com.fasterxml.jackson.core.base.ParserMinimalBase._reportUnexpectedChar(ParserMinimalBase.java:635)
	at com.fasterxml.jackson.core.json.ReaderBasedJsonParser._handleOddName(ReaderBasedJsonParser.java:1814)
	at com.fasterxml.jackson.core.json.ReaderBasedJsonParser.nextFieldName(ReaderBasedJsonParser.java:941)
	at com.fasterxml.jackson.databind.deser.std.BaseNodeDeserializer.deserializeObject(JsonNodeDeserializer.java:268)
	at com.fasterxml.jackson.databind.deser.std.JsonNodeDeserializer.deserialize(JsonNodeDeserializer.java:69)
	at com.fasterxml.jackson.databind.deser.std.JsonNodeDeserializer.deserialize(JsonNodeDeserializer.java:16)
	at com.fasterxml.jackson.databind.deser.DefaultDeserializationContext.readRootValue(DefaultDeserializationContext.java:322)
	at com.fasterxml.jackson.databind.ObjectMapper._readTreeAndClose(ObjectMapper.java:4635)
	at com.fasterxml.jackson.databind.ObjectMapper.readTree(ObjectMapper.java:3042)
	at io.hyperfoil.tools.horreum.svc.Util.toJsonNode(Util.java:184)
	at io.hyperfoil.tools.horreum.svc.RunServiceImpl.transform(RunServiceImpl.java:1142)
	...
```
because you are trying to deserialize string
```
{outcome: {foo: 42}}
```
And this string was created using `result = value.toString();` on line 1127.
The result of the `Value::toString()` is not a valid JSON. Hence, please don't use that method to obtain something that is not a valid JSON.

Which part am I wrong in?

Reply:
Subsequently, I've changed the code to
```
JsonNode result = null;
                        ByteArrayOutputStream out = new ByteArrayOutputStream();
                        try (org.graalvm.polyglot.Context context = org.graalvm.polyglot.Context.newBuilder("js").out(out).err(out).build()) {
                           context.enter();
                           try {
                              Value value = context.eval("js", code.toString());
                              result = Util.convertToJson(value);
                 ...
                        if (result != null) {
                           JsonNode node = result;
```
and the test (stil without `JSON.stringify`) passes.

Reply:
thanks for pointing that out.

Reply:
Does it produce a TextNode when you remove all uses of `JSON.stringify(...)`?